src/Parsing/String.purs

```diff
@@ -231,14 +234,24 @@ match p = do
 -- |
 -- | #### Example
 -- |
--- | This example shows how to compile and run the `xMany` parser which will
--- | capture the regular expression pattern `x*`.
+-- | Compiling and running different regex parsers:
 -- |
 -- | ```purescript
--- | case regex "x*" noFlags of
--- |   Left compileError -> unsafeCrashWith $ "xMany failed to compile: " <> compileError
--- |   Right xMany -> runParser "xxxZ" do
--- |     xMany
+-- | example re flags text =
```

Comment:
The first example should show the simplest basic usage. Adding this `example` helper function is an extra indirection.

Reply:
But the code has always been there, I just moved it to a separate function. I can move the function after example if you want.